photutils/02_photutils_source_detection.ipynb

```diff
@@ -0,0 +1,1034 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from photutils import EllipticalAperture"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the approximate isophotal extent\n",
+    "r = 4.\n",
+    "\n",
+    "# Create the apertures\n",
+    "apertures = []\n",
+    "for obj in catalog:\n",
+    "    position = (obj.xcentroid.value, obj.ycentroid.value)\n",
+    "    a = obj.semimajor_axis_sigma.value * r\n",
+    "    b = obj.semiminor_axis_sigma.value * r\n",
+    "    theta = obj.orientation.value\n",
+    "    apertures.append(EllipticalAperture(position, a, b, theta=theta))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up the figure with subplots\n",
+    "fig, ax1 = plt.subplots(1, 1, figsize=(8, 8))\n",
+    "\n",
+    "# Plot the data\n",
+    "fitsplot = ax1.imshow(np.ma.masked_where(xdf_image.mask, xdf_image_clipped), norm=norm_image)\n",
+    "\n",
+    "# Plot the apertures\n",
+    "for aperture in apertures:\n",
+    "    aperture.plot(color='red', lw=1, alpha=0.7, ax=ax1)\n",
+    "\n",
+    "# Define the colorbar\n",
+    "cbar = plt.colorbar(fitsplot, fraction=0.046, pad=0.04, ticks=LogLocator(subs=range(10)))\n",
+    "labels = ['$10^{-4}$'] + [''] * 8 + ['$10^{-3}$'] + [''] * 8 + ['$10^{-2}$']\n",
+    "cbar.ax.set_yticklabels(labels)\n",
+    "\n",
+    "# Define labels\n",
+    "cbar.set_label(r'Flux Count Rate ($e^{-1}/s$)', rotation=270, labelpad=30)\n",
+    "ax1.set_xlabel('X (pixels)')\n",
+    "ax1.set_ylabel('Y (pixels)')\n",
+    "ax1.set_title('Segmentation Image Apertures')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<h3>**Exercises:**</h3><br>\n",
+    "Play with the isophotal extent of the elliptical apertures (defined above as `r`). Observe how changing this value affects the apertures that are created.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is clear that using `photutils` for image segmentation can allow users to generate highly customized apertures - great for complex data that contain many different kinds of celestial sources."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Conclusions\n",
+    "\n",
+    "The `photutils` package provides users with a variety of methods for detecting sources in their data, from familar algorithms such as `DAOFind` and `starfind`, to more complex and customizable image segmentation algorithms. These methods allow for easy creation of a diverse array of apertures that can be used for photometric analysis.\n",
```

Comment:
Maybe there should also be a "and if you want to write your own algorithm, you can do that too!" at the end here? One or both of these could be mentioned:

* The photutils docs are not as fleshed out on this as might be desirable, but the basic idea is that anything that subclasses this: https://photutils.readthedocs.io/en/latest/api/photutils.detection.StarFinderBase.html#photutils.detection.StarFinderBase can be used in the exact same way.
* Photutils is open developed so PRs implementing new star finders are always welcome.